integrations/kube-agent-updater/pkg/version/basichttp.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/integrations/kube-agent-updater/pkg/cache"
+)
+
+const (
+	// versionPath is the version discovery endpoint returning the current
+	// target version as defined in
+	// https://github.com/gravitational/teleport/blob/master/rfd/0109-cloud-agent-upgrades.md#version-discovery-endpoint
+	versionPath   = "version"
+	httpTimeout   = 10 * time.Second
+	cacheDuration = time.Minute
+)
+
+// basicHTTPVersionClient retrieves the version from an HTTP endpoint
+// it should not be invoked immediately and must be wrapped in a cache layer in
+// order to avoid spamming the version server in case of reconciliation errors.
+// use BasicHTTPVersionGetter if you need to get a version.
+type basicHTTPVersionClient struct {
+	baseURL *url.URL
+	client  *http.Client
+}
+
+// Get sends an HTTP GET request and returns the version prefixed by "v".
+// It expects the endpoint to be unauthenticated, return 200 and the response
+// body to contain a valid semver tag without the "v".
+func (b *basicHTTPVersionClient) Get(ctx context.Context) (string, error) {
```

Comment:
Looks like this is 90% the same as "basichttp" from maintenance client, would it make sense to move common parts out into a generic "http client" and reuse in both places to reduce code duplication?